### PR TITLE
Linter - fix ignoring input parameter for constructor

### PR DESCRIPTION
**Pull request by keradus**
Linter - fix ignoring input parameter for constructor.
Also, add some tests.

### CODE REVIEW

File: Symfony/CS/Tests/Linter/LinterTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public function testLintSourceWithGoodCode()
+    {
+        $linter = new Linter();
+        $linter->lintSource('<?php echo 123;'); // no exception should be raised
+    }
+
+    /**
+     * @covers Symfony\CS\Linter\Linter::lintSource
+     *
+     * @expectedException Symfony\CS\Linter\LintingException
```
Comment (GrahamCampbell) [inline Symfony/CS/Tests/Linter/LinterTest.php]: We should definitely test the exception message too.

Comment (keradus) [inline Symfony/CS/Tests/Linter/LinterTest.php]: the message depends on PHP version, and here is a problem with Windows vs Unix

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Linter/LinterTest.php]: Yes, indeed, but we can use phpunits regex message matching

Comment (keradus) [inline Symfony/CS/Tests/Linter/LinterTest.php]: done, let see how it will behave

File: Symfony/CS/Linter/Linter.php [hunk 2]
```diff
@@ -40,17 +40,17 @@
...
     {
-        if (null === $phpExecutable) {
+        if (null === $executable) {
             $executableFinder = new PhpExecutableFinder();
             $executable = $executableFinder->find();
-        }
 
-        if (empty($executable)) {
-            throw new UnavailableLinterException();
+            if (empty($executable)) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Linter/Linter.php]: why are we calling a function instead of doing a comparison?

Comment (keradus) [inline Symfony/CS/Linter/Linter.php]: dunno, the logic was already here

**Pull request closed without merging**